app/code/Magento/Sitemap/Model/Sitemap.php

```diff
@@ -192,30 +187,36 @@ class Sitemap extends \Magento\Framework\Model\AbstractModel implements \Magento
-     * @param ResourceModel\Catalog\CategoryFactory $categoryFactory
-     * @param ResourceModel\Catalog\ProductFactory $productFactory
-     * @param ResourceModel\Cms\PageFactory $cmsFactory
-     * @param \Magento\Framework\Stdlib\DateTime\DateTime $modelDate
-     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
-     * @param \Magento\Framework\App\RequestInterface $request
-     * @param \Magento\Framework\Stdlib\DateTime $dateTime
-     * @param \Magento\Framework\Model\ResourceModel\AbstractResource|null $resource
-     * @param \Magento\Framework\Data\Collection\AbstractDb|null $resourceCollection
-     * @param array $data
-     * @param DocumentRoot|null $documentRoot
-     * @param ItemProviderInterface|null $itemProvider
-     * @param SitemapConfigReaderInterface|null $configReader
+     * @param \Magento\Framework\Model\Context                        $context
+     * @param \Magento\Framework\Registry                             $registry
+     * @param ExtensionAttributesFactory                              $extensionFactory
+     * @param AttributeValueFactory                                   $customAttributeFactory
+     * @param \Magento\Framework\Model\ResourceModel\AbstractResource $resource
+     * @param \Magento\Framework\Data\Collection\AbstractDb|null      $resourceCollection
+     * @param \Magento\Framework\Escaper                              $escaper
+     * @param \Magento\Sitemap\Helper\Data                            $sitemapData
+     * @param \Magento\Framework\Filesystem                           $filesystem
+     * @param ResourceModel\Catalog\CategoryFactory                   $categoryFactory
+     * @param ResourceModel\Catalog\ProductFactory                    $productFactory
+     * @param ResourceModel\Cms\PageFactory                           $cmsFactory
+     * @param \Magento\Framework\Stdlib\DateTime\DateTime             $modelDate
+     * @param \Magento\Store\Model\StoreManagerInterface              $storeManager
+     * @param \Magento\Framework\App\RequestInterface                 $request
+     * @param \Magento\Framework\Stdlib\DateTime                      $dateTime
+     * @param array                                                   $data
+     * @param DocumentRoot|null                                       $documentRoot
+     * @param ItemProviderInterface|null                              $itemProvider
+     * @param SitemapConfigReaderInterface|null                       $configReader
      * @param \Magento\Sitemap\Model\SitemapItemInterfaceFactory|null $sitemapItemFactory
+     *
+     * @throws \Magento\Framework\Exception\FileSystemException
      * @SuppressWarnings(PHPMD.ExcessiveParameterList)
      */
     public function __construct(
         \Magento\Framework\Model\Context $context,
         \Magento\Framework\Registry $registry,
+        ExtensionAttributesFactory $extensionFactory,
+        AttributeValueFactory $customAttributeFactory,
```

Comment:
Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

Reply:
Am I wrong in thinking that I need to make the object extend AbstractExtensibleModel? If so, how do I make these changes backwards compatible, seeing how they require completely different things in the constructor? Just use the object manager for everything?